aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -115,6 +118,24 @@ public void close() {
         group.shutdownGracefully();
     }
 
+    /**
+     * Depending on the EventLoopGroup used we may need to use a different socket channel.
+     */
+    @ReviewBeforeRelease("Perhaps we should make the customer provide both event loop group" +
+                         "and channel in some kind of wrapper class to avoid having to do this.")
+    private Class<? extends Channel> resolveSocketChannelClass() {
```

Comment:
This would also help unify the 'if this is available' logic - so you don't have to repeat it.